smithy-model/src/main/java/software/amazon/smithy/model/knowledge/IdentifierBindingIndex.java

```diff
@@ -155,25 +152,29 @@ private void processResource(ResourceShape resource, OperationIndex operationInd
     }
 
     private boolean isCollection(ResourceShape resource, ToShapeId operationId) {
-        return resource.getCollectionOperations().contains(operationId)
+        return resource.getCollectionOperations().contains(operationId.toShapeId())
```

Comment:
Note: I changed this too because this was raising a warning because `operationId` is a `ToShapeId` but the set contains `ShapeId`.